lib/msf/core/exploit/remote/nuuo.rb

```diff
@@ -0,0 +1,232 @@
+require 'msf/core/exploit/tcp'
+
+###
+#
+# This module exposes methods that may be useful to exploits that deal with
+# servers that speak the Nuuo protocols for their devices and management software.
+#
+###
+module Msf
+module Exploit::Remote::Nuuo
+  include Exploit::Remote::Tcp
+
+  #
+  # Creates an instance of an Nuuo exploit module.
+  #
+  def initialize(info = {})
+    super(update_info(info,
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib@gmail.com>'
+        ],
+    ))
+
+    register_options(
+      [
+        Opt::RHOST,
+        Opt::RPORT(5180),
+        OptString.new('SESSION', [false, 'Session number of logged in user']),
+        OptString.new('USERNAME', [true, 'Username to login as', 'admin']),
+        OptString.new('PASSWORD', [false, 'Password for the specified user']),
+      ], Msf::Exploit::Remote::Nuuo)
+
+    register_advanced_options(
+      [
+        OptString.new('PROTOCOL', [ true, 'Nuuo protocol', 'NUCM/1.0']),
+      ], Msf::Exploit::Remote::Ftp)
+
+    @session = nil
+
+    # All CMS versions at time of release
+    # Note that these primitives are not guaranteed to work in all versions
+    # Add new version strings here
+    @versions =
+      [
+        "1.3.1",
+        "1.3.3",
+        "1.5.0",
+        "1.5.2",
+        "1.6.0",
+        "1.7.0",
+        "2.1.0",
+        "2.3.0",
+        "2.3.1",
+        "2.3.2",
+        "2.4.0",
+        "2.5.0",
+        "2.6.0",
+        "2.7.0",
+        "2.8.0",
+        "2.9.0",
+        "2.10.0",
+        "2.11.0",
+        "3.0.0",
+        "3.1.0",
+        "3.2.0",
+        "3.3.0",
+        "3.4.0",
+        "3.5.0"
+      ]
+
+    @version = nil
+  end
+
+
+  ##
+  # Formats the message we want to send into the correct protocol format
+  ##
+  def format_msg(msg)
```

Comment:
Suggest `nuuo_format_msg()`.

Do you expect to use `format_msg()` outside of the mixin?  If not this should be made private in this module.

See https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/http/joomla/version.rb for example mixin using private methods.